shinken/objects/service.py

```diff
@@ -1295,18 +1301,90 @@ def add_item(self, item, index=True):
             in_file = " in %s" % source
         else:
             in_file = ""
-        if not hname and not hgname:
+        if not hname and not hgname and not sdesc:
             mesg = "a %s has been defined without host_name nor " \
-                   "hostgroups%s" % (objcls, in_file)
-            item.configuration_errors.append(mesg)
-        if not sdesc:
-            mesg = "a %s has been defined without service_description%s" % \
-                   (objcls, in_file)
+                   "hostgroups nor service_description%s" % (objcls, in_file)
             item.configuration_errors.append(mesg)
+        elif not sdesc or sdesc and not hgname and not hname and not was_partial:
+            self.add_partial_service(item, index, (objcls, hname, hgname, sdesc, in_file))
+            return
+
         if index is True:
             item = self.index_item(item)
         self.items[item.id] = item
 
+    def add_partial_service(self, item, index=True, var_tuple=None):
+        if var_tuple is None:
+            return
+
+        objcls, hname, hgname, sdesc, in_file = var_tuple
+        use = getattr(item, 'use', [])
+
+
+        if use == []:
+            mesg = "a %s has been defined without host_name nor " \
+                   "hostgroups nor service_description and " \
+                   "there is no use to create a unique key%s" % (objcls, in_file)
+            item.configuration_errors.append(mesg)
+            return
+
+        use = ','.join(use)
+        if sdesc:
+            name = "::".join((sdesc, use))
+        elif hname:
+            name = "::".join((hname, use))
+        else:
+            name = "::".join((hgname, use))
+
+        if name in self.name_to_partial:
+            item = self.manage_conflict(item, name, partial=True)
+        self.name_to_partial[name] = item
+
+        self.partial_services[item.id] = item
+
+    # Inheritance for just a property
+    def apply_partial_inheritance(self, prop):
+        for i in itertools.chain(self.items.itervalues(),
+                                 self.partial_services.itervalues(),
+                                 self.templates.itervalues()):
+            i.get_property_by_inheritance(prop)
+            # If a "null" attribute was inherited, delete it
+            try:
+                if getattr(i, prop) == 'null':
+                    delattr(i, prop)
+            except AttributeError:
+                pass
+
+    def apply_inheritance(self):
```

Comment:
it's quite possible to re-use/extend its parent `Items.apply_inheritance()` function.
and nearly only have in this one :
```
for i in self.partial_services.itervalues():
            self.add_item(i, True, True)
```

Same comment for `linkify_templates()`.

otherwise it's yet another code duplication but 5% different  from the Items one.. :s